src/java/org/apache/cassandra/db/CounterMutationCallback.java

```diff
@@ -40,12 +39,14 @@ public class CounterMutationCallback implements Runnable
 {
     private final Message<CounterMutation> requestMessage;
     private final InetAddressAndPort respondToAddress;
+    private final RequestSensors sensors;
     private int replicaCount = 0;
 
-    public CounterMutationCallback(Message<CounterMutation> requestMessage, InetAddressAndPort respondToAddress)
+    public CounterMutationCallback(Message<CounterMutation> requestMessage, InetAddressAndPort respondToAddress, RequestSensors sensors)
```

Comment:
`RequestSensorsFactory.instance.create()` is and should be invoked at the `VerbHandler` https://github.com/datastax/cassandra/blob/88a88116349eaf717a742cd5ba64998b931d10b1/src/java/org/apache/cassandra/db/CounterMutationVerbHandler.java#L52

Having said that, I can keep this code as before this patch by consulting `RequestTracker` but I was thinking to:
1. Either pass `SensorsRequest` around, from the code that initilized it, and avoid nullability check
2. Consulting `RequestTracker` but it is better to check for nullability because in principle the `ExecutorLocal#get` is nullable by design

Reply:
I see what you mean. It's just that passing it around _seems_ to diminish readability, as we now have `RequestSensors` and `Collection<Sensor>`, and oddly enough the former is often called `sensors` and the latter `requestSensors`, so maybe some renaming would help? Otherwise I'll leave it to you.